src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -25,12 +30,26 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public string GetCommandLine(string inputFile, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            string format = GetDocumentTypeString(publishFormat);
-            string outputFile = Path.GetFileName(outputFilePath);
-            string outputFolder = Path.GetDirectoryName(outputFilePath).Replace('\\', '/');
-            // Run rmarkdown::render
-            return Invariant($"rmarkdown::render('{inputFile}', output_format='{format}', output_file='{outputFile}',  output_dir='{outputFolder}', encoding='cp{encoding.CodePage}')");
+        public async Task PublishAsync(IRSession session, ICoreShell coreShell,  IFileSystem fs, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            try {
+                using(var fts = new FileTransferSession(session, fs)) {
+                    var rmd = await fts.SendFileAsync(inputFilePath);
+                    var result = await RMarkdownRenderAsync(session, rmd, outputFilePath, publishFormat, encoding);
+                    await fts.FetchFileAsync(result, outputFilePath);
```

Comment:
We currently compute `outputFilePath` as `inputFilePath` with a different extension (see the call site for `PublishAsync`). But this only works when both input and output files are on the same local filesystem (indeed, we get the input path from the text buffer, which will always be local). As soon as local and remote are different, this will break. 

Furthermore, we currently use `outputFilePath` for two completely different things. First, we pass it to `RMarkdownRenderAsync`, which will use it as a destination on the remote file system to write the output to. Then we fetch the produced file as a blob, and write it to `outputFilePath` onto the local file system. That doesn't make sense to me.

It feels like the remote path is actually irrelevant, and should just be another temp filename, like the input `rmd`, since we immediately load that file into a blob anyway.